splipy/io/ofoam.py

```diff
@@ -0,0 +1,115 @@
+from itertools import groupby
+import numpy as np
+from operator import itemgetter
+from os.path import exists, isdir, join
+from os import makedirs
+
+from splipy import SplineModel
+
+
+class OpenFOAM(object):
+
+    def __init__(self, target):
+        self.target = target
+
+    def __enter__(self):
+        # Create the target directory if it does not exist
+        if not exists(self.target):
+            makedirs(target)
+        # If it does, ensure that it's a directory
+        elif not isdir(self.target):
+            raise FileExistsError('{} exists and is not a directory'.format(self.target))
+
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        pass
+
+    def _header(self, cls, obj, note=None):
+        s = 'FoamFile\n{\n'
+        s += '    version     2.0;\n'
+        s += '    format      ascii;\n'
+        s += '    class       %s;\n' % cls
+        if note:
+            s += '    note        "%s";\n' % note
+        s += '    object      %s;\n' % obj
+        s += '}\n'
+        return s
+
+    def write(self, model):
+        assert isinstance(model, SplineModel)
+
+        # Only linear volumes in 3D, please
+        assert model.pardim == 3
```

Comment:
Would it make sense to make the default behavior of `volume_factory.extrude()` of length 1 in the z-direction for all surface models?

Reply:
Yes, but I wanted the first version to not do too many things without the user asking.

Reply:
Makes sense. And it really isn't too much to ask of the user.